lib/concurrent/executor/safe_task_executor.rb

```diff
@@ -6,17 +6,15 @@ module Concurrent
   # success - indicating if the callable has been executed without errors
   # value - filled by the callable result if it has been executed without errors, nil otherwise
   # reason - the error risen by the callable if it has been executed with errors, nil otherwise
-  class SafeTaskExecutor
+  class SafeTaskExecutor < Synchronization::Object
 
     def initialize(task, opts = {})
-      @task = task
-      @mutex = Mutex.new
-      @exception_class = opts.fetch(:rescue_exception, false) ? Exception : StandardError
+      super(task, opts)
     end
```

Comment:
I think he is referring to the fact that inside a synchornization object, ivars assigned in camel case are final, and will never be changed. i.e.:

```ruby
@Mutex = Mutex.new
```